src/main/scala/sigmastate/serialization/transformers/ProveDiffieHellmanTupleSerializer.scala

```diff
@@ -18,18 +18,45 @@ case class ProveDiffieHellmanTupleSerializer(cons:
 
   override val opCode: OpCode = OpCodes.ProveDiffieHellmanTupleCode
 
-  override def serializeBody(obj: ProveDiffieHellmanTuple, w: ByteWriter): Unit = {
-    w.putValue(obj.gv)
-    w.putValue(obj.hv)
-    w.putValue(obj.uv)
-    w.putValue(obj.vv)
+  private val constCodePrefix: Byte = 0
+
+  override def serializeBody(obj: ProveDiffieHellmanTuple, w: ByteWriter): Unit = obj match {
+    case ProveDiffieHellmanTuple(Constant(_, _), Constant(_, _), Constant(_, _), Constant(_, _)) =>
+      val tpe = obj.gv.tpe
+      assert(tpe == obj.hv.tpe && tpe == obj.uv.tpe && tpe == obj.vv.tpe,
+        s"constants with the same tpe are expected, got $obj")
+      w.put(constCodePrefix)
+      w.putType(tpe)
+      DataSerializer.serialize(obj.gv.asInstanceOf[GroupElementConstant].value, tpe, w)
```

Comment:
@greenhat why not extract the value in the pattern?

Reply:
Yeah, not good enough. Fixed. Thanks.